lib/processor.js

```diff
@@ -252,7 +252,9 @@ function preprocess(text) {
     });
 
     return blocks.map((block, index) => ({
-        filename: `${index}.${block.lang}`,
+
+        // TODO: Use trimStart() after dropping Node.js 8.x support.
+        filename: `${index}.${block.lang.replace(/^\s+/u, "").split(" ")[0]}`,
```

Comment:
I don't think you need the `/u` regex flag but can't you just use `trim()` since there is no reason to keep trailing spaces?

`${index}.${block.lang.trim().split(" ")[0]}`

Reply:
Good idea, there's no reason this _has_ to trim only the leading whitespace. I went ahead and added a test for trailing whitespace while I was at it.